shell.c

```diff
@@ -306,16 +309,94 @@ char* readLine() {
     }
 }
 
+// Display result of the assignment
+struct pre_assessment {
+    bool valid;
+    char *assignment;
+};
+
+void display_assignment_result(struct pre_assessment pa, char *line) {
+    // Retrieve the first 8 characters forming the "echo -e "
+    const int COMM_SIZE = 8;
+    char command[COMM_SIZE + 1];
+    for (int i = 0; i < COMM_SIZE; i++) command[i] = line[i];
+    command[COMM_SIZE] = '\0';
+
+    // Retrieve the string inside the quotation marks
+    char display_str[strlen(line)];
+    for (int i = (COMM_SIZE + 1); i < (strlen(line)-1); i++) display_str[i-(COMM_SIZE + 1)] = line[i];
+
+    // Retrieve index of '=' sign
+    int start_index = 0;
+    for (int i = 0; i < strlen(pa.assignment); i++) {
+        if (pa.assignment[i] == '=') {
+            start_index = i;
+            break;
+        }
+    }
+
+    // Extract value after '=' sign
+    char value_str[strlen(pa.assignment)];
+    for (int i = start_index+1; i < strlen(pa.assignment); i++) {
+        value_str[i-(start_index+1)] = pa.assignment[i];
+    }
+
+    // 
+
+    if (!strcmp(command, "echo -e ")) printf("%s%s\n", display_str, value_str);
+    // return false;
+}
+
+// Shell substitution
+struct pre_assessment verify_assignment_syntax(char *line) {
+    /* Validate:
+        - Variable is followed by "="
+        - Only 1 "=" in the assignment
+    */
+
+    int index = 0;
+    int equal_sign_count = 0;
+    struct pre_assessment a;
+
+    while (index < strlen(line)) {
+        if (line[index] == '=') equal_sign_count += 1;
+        index += 1;
+    }
+
+    a.assignment = "";
+    a.valid = false;
+    if (equal_sign_count == 1) {
+        a.assignment = line;
+        a.valid = true;
+    }
+    return a;
+    // printf("command not found\n");
+    // exit(0);
+}
+
 void loop() {
     char *line;
     struct job *job;
 
-    while(1) {
+    struct pre_assessment assessment;
+    assessment.assignment = "";
+    assessment.valid = false;
```

Comment:
Maybe variable initialization should be in init function.

Reply:
I will take a look at this. Because I got an error thus I had to initialize somewhere to get it working.